### DISCUSSION TITLE
Trouble with preview overlays

### DISCUSSION THREAD

**Discussion by MikeLerwill99** [raspberrypi/picamera2#580] [Q&A] [answered]
The following code is intended to show a preview window of the entire camera image, which it does. It then adds an overlay which should be a 900x900 square over the centre of the image, the actual result is a rectangle that is 3 times as wide as it is high but it is centred over the camera image

import time
import numpy as np

from picamera2 import Picamera2, Preview

picam2 = Picamera2()
picam2.configure(picam2.create_preview_configuration({"size": (2304,1296)}))
picam2.start_preview(Preview.QTGL, width = int (2304/2), height = int (1296/2))
picam2.start()
time.sleep(1)

# Create an array the same size as the image
overlay = np.zeros((2304,1296,4), dtype=np.uint8)
# Fill a 900x900 pixel square centred with blueish
overlay[702:1602, 198:1098] = (0, 0, 255, 64)
# Set it as an overlay 
picam2.set_overlay(overlay)
time.sleep(20)

Can anyone help with what I may be doing wrong.

regards

Mike Lerwill

**Comment by davidplowman** [comment, marked as answer]
Hi, I haven't had time to try out your example, but my first thought is just to check the order of your numpy dimensions. Remember, numpy is height first, width second. If that doesn't sort it out please post back. Thanks!

**Reply by MikeLerwill99** [reply]
David,
Thank you for taking the time to reply, that was exactly my problem and I would never have got there without your help.

regards
Mike Lerwill
